src/execution/compiler/operator/analyze_translator.cpp

```diff
@@ -0,0 +1,403 @@
+      pg_statistic_table_schema_(catalog::postgres::Builder::GetStatisticTableSchema()),
+      pg_statistic_table_pm_(GetCodeGen()
+                                 ->GetCatalogAccessor()
+                                 ->GetTable(catalog::postgres::PgStatistic::STATISTIC_TABLE_OID)
+                                 ->ProjectionMapForOids(std::vector<catalog::col_oid_t>(
+                                     catalog::postgres::PgStatistic::PG_STATISTIC_ALL_COL_OIDS.begin(),
+                                     catalog::postgres::PgStatistic::PG_STATISTIC_ALL_COL_OIDS.end()))),
+      pg_statistic_index_schema_(catalog::postgres::Builder::GetStatisticOidIndexSchema(plan.GetDatabaseOid())),
+      pg_statistic_index_pm_(GetCodeGen()
+                                 ->GetCatalogAccessor()
+                                 ->GetIndex(catalog::postgres::PgStatistic::STATISTIC_OID_INDEX_OID)
+                                 ->GetKeyOidToOffsetMap()),
+      pg_statistic_col_oids_(GetCodeGen()->MakeFreshIdentifier("pg_statistic_col_oids")),
+      table_oid_(GetCodeGen()->MakeFreshIdentifier("table_oid")),
+      col_oid_(GetCodeGen()->MakeFreshIdentifier("col_oid")),
+      num_rows_(GetCodeGen()->MakeFreshIdentifier("num_rows")),
+      pg_statistic_index_iterator_(GetCodeGen()->MakeFreshIdentifier("pg_statistic_index_iterator")),
+      pg_statistic_index_pr_(GetCodeGen()->MakeFreshIdentifier("pg_statistic_index_pr")),
+      pg_statistic_updater_(GetCodeGen()->MakeFreshIdentifier("pg_statistic_updater")),
+      pg_statistic_update_pr_(GetCodeGen()->MakeFreshIdentifier("pg_statistic_update_pr")) {
+  // Analyze is serial
+  pipeline->RegisterSource(this, Pipeline::Parallelism::Serial);
+  // Prepare child
+  compilation_context->Prepare(*plan.GetChild(0), pipeline);
+  auto *codegen = GetCodeGen();
+
+  pg_statistic_column_lookup_[catalog::postgres::PgStatistic::STARELID.oid_] = table_oid_;
+  pg_statistic_column_lookup_[catalog::postgres::PgStatistic::STAATTNUM.oid_] = col_oid_;
+  pg_statistic_column_lookup_[catalog::postgres::PgStatistic::STA_NUMROWS.oid_] = num_rows_;
+
+  for (const auto &col_info : catalog::postgres::PgStatistic::ANALYZE_AGGREGATES) {
+    auto agg_var = codegen->MakeFreshIdentifier("stat_col_" + std::to_string(col_info.column_oid_.UnderlyingValue()));
+    aggregate_variables_.emplace_back(agg_var);
+    pg_statistic_column_lookup_[col_info.column_oid_] = agg_var;
+  }
+}
+
+void AnalyzeTranslator::PerformPipelineWork(WorkContext *context, FunctionBuilder *function) const {
+  const auto &plan = GetPlanAs<planner::AnalyzePlanNode>();
+  auto *codegen = GetCodeGen();
+
+  NOISEPAGE_ASSERT(plan.GetChild(0)->GetOutputSchema()->NumColumns() ==
+                       catalog::postgres::PgStatistic::NUM_ANALYZE_AGGREGATES * plan.GetColumnOids().size() + 1,
+                   fmt::format("We should 1 aggregate for the number of rows and then {} aggregates per column",
```

Comment:
No worries, English is tougher when Java is your first language.

Reply:
Fixed